presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestEliminateCrossJoins.java

```diff
@@ -114,9 +114,16 @@ public void testEliminateCrossJoinWithNonEqualityCondition()
                         join(INNER, ImmutableList.of(equiJoinClause("L_ORDERKEY", "O_ORDERKEY")),
                                 anyTree(
                                         join(INNER, ImmutableList.of(equiJoinClause("P_PARTKEY", "L_PARTKEY")), Optional.of("P_NAME < cast(L_COMMENT AS varchar(55))"),
-                                                anyTree(PART_WITH_NAME_TABLESCAN),
-                                                anyTree(filter("L_PARTKEY <> L_ORDERKEY", LINEITEM_WITH_COMMENT_TABLESCAN)))),
-                                anyTree(ORDERS_TABLESCAN))));
+                                                anyTree(
+                                                        filter("NOT(P_PARTKEY IS NULL)",
+                                                                PART_WITH_NAME_TABLESCAN)),
+                                                anyTree(
+                                                        filter(
+                                                                "NOT(L_ORDERKEY IS NULL) AND NOT(L_PARTKEY IS NULL) AND L_PARTKEY <> L_ORDERKEY",
```

Comment:
Add ``// TODO we should eliminate NULL check on L_ORDERKEY and L_PARTKEY, because of `<>` predicate on them``